File: regions/system/library/swift/tracing/Sources/K1s0Tracing/Baggage.swift
public struct Baggage: Sendable {
    private var entries: [String: String] = [:]

    public init() {}

    public mutating func set(_ key: String, _ value: String) {
        entries[key] = value
    }

    public func get(_ key: String) -> String? {
        entries[key]
    }

    public func toHeader() -> String {
        entries.map { "\($0.key)=\($0.value)" }.joined(separator: ",")
    }

    public static func fromHeader(_ s: String) -> Baggage {
        var baggage = Baggage()
        guard !s.isEmpty else { return baggage }
        for item in s.split(separator: ",") {
            let parts = item.split(separator: "=", maxSplits: 1)
            if parts.count == 2 {
                let key = parts[0].trimmingCharacters(in: .whitespaces)
                let value = parts[1].trimmingCharacters(in: .whitespaces)
                baggage.set(key, value)
            }
        }
        return baggage
    }

    public var isEmpty: Bool {
        entries.isEmpty
    }
}

File: regions/system/library/python/tracing/src/k1s0_tracing/baggage.py
"""W3C Baggage propagation."""

from __future__ import annotations


class Baggage:
    """W3C Baggage container."""

    def __init__(self) -> None:
        self._entries: dict[str, str] = {}

    def set(self, key: str, value: str) -> None:
        self._entries[key] = value

    def get(self, key: str) -> str | None:
        return self._entries.get(key)

    def to_header(self) -> str:
        return ",".join(f"{k}={v}" for k, v in self._entries.items())

    @classmethod
    def from_header(cls, s: str) -> Baggage:
        b = cls()
        if not s:
            return b
        for item in s.split(","):
            if "=" in item:
                k, _, v = item.partition("=")
                b.set(k.strip(), v.strip())
        return b

    @property
    def is_empty(self) -> bool:
        return len(self._entries) == 0
